core/domain/html_cleaner.py

```diff
@@ -790,3 +800,46 @@ def add_caption_attr_to_image(html_string):
             image['caption-with-value'] = escape_html(json.dumps(''))
 
     return unicode(soup)
+
+
+def add_dimensions_to_noninteractive_image_tag(html_string, exp_id):
+    """Adds dimensions to all oppia-noninteractive-image tags.
+
+    Args:
+        html_string: str. HTML string in which the dimensions is to be
+            added.
+        exp_id: str. Exploration id.
+
+    Returns:
+        str. Updated HTML string with the dimensions for all
+            oppia-noninteractive-image tags.
+    """
+    soup = bs4.BeautifulSoup(html_string.encode('utf-8'), 'html.parser')
+
+    for image in soup.findAll('oppia-noninteractive-image'):
+        filename = unescape_html(image['filepath-with-value'])
+        filename = filename[1:-1]
+        image['filepath-with-value'] = escape_html(json.dumps(
+            get_filepath_of_object_image(filename, exp_id)))
+    return unicode(soup)
+
+
+def get_filepath_of_object_image(filename, exp_id):
+    """Gets the dimensions of the image file.
+
+    Args:
+        filename: str. Name of the file whose dimensions need to be
+            calculated.
+        exp_id: str. Exploration id.
+
+    Returns:
+        object. filepath object of the image.
+    """
+    URL = ('https://storage.googleapis.com/%s/%s/assets/image/%s' % (
+        app_identity_services.get_gcs_resource_bucket_name(), exp_id,
+        filename)) if not feconf.DEV_MODE else (
+            'http://localhost:8181/imagehandler/%s/%s' % (exp_id, filename))
+    imageFile = cStringIO.StringIO(urllib.urlopen(URL).read())
+    img = Image.open(imageFile)
+    width, height = img.size
```

Comment:
Shouldn't you close the image? (I'm not sure)

Reply:
No, it does that automatically